archesproject/arches-for-science-prj
# Mark strings for localization throughout the project #1503

**Changes**
- Mark strings for localization throughout the project.
- Fix #1269
- Follow-up to #1262, looks like some strings were missed there
- [Rename "region" to "analysis area"](https://github.com/archesproject/arches-for-science/pull/1515/commits/bc66efe164eb4e77eeeb02766a7058db98301358) https://github.com/archesproject/arches-for-science/issues/1504
- Identify follow-ups with `TODO(i18n)`:
  - #1516
  - #1518
  - #1519
  - #1520

There's a start toward the last item in b452d6bc17deb842a0358b1c8ff87034b9648d68.

**Other stuff:**
- Require arches 7.6 (since this PR assumes archesproject/arches#10498 is going in)
- Add LOCALE_PATHS to settings (so that .po files can be generated)
   - To look at the .po files:
```
cd arches-for-science/arches_for_science
PYTHONPATH='../../arches' python3 ../manage.py makemessages --local=nl --settings='arches_for_science.test_settings' --extension=htm,py
```
The PYTHONPATH is because arches.tests is not importable otherwise.

cd'ing to the app directory makes it easier to ignore build/ and node_modules/ files without having to specify that on the CLI, but it produces slightly different output than running from the project dir, so I'll defer to @aarongundel on how we're going to do that in prod. But this works for conjuring up a .po file to look at. I don't know if we're only going to generate files for disco, or if we're going to generate them for AfS and commit to the AfS repo.


- PR contains #1513 fix, but will rebase that out later when it's merged on its own
- I took the liberty of standardizing a couple report table headers and error messages where the slight discrepancies would have created unnecessary work for translators IMO.
- Since I was on the hook 🪝 for combing through every file anyway and checking that aria-labels were marked for translation, I took the liberty of adding `aria-hidden="true"` on decorative icons. [More at MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden).

**Testing instructions:**
It's too much to test, but sanity checking a few workflows would be good.
Make sure you check out archesproject/arches#10498 before testing.
To reset your db:
```
python3 manage.py packages -o load_package -a arches_for_science -db -dev
python3 manage.py packages -o load_package -s <path to afs-test-pkg/pkg>
```
You might also enable `'nl'` in LANGUAGES in disco/settings.py

You could also generate the .po files as demo'd above and glance over the strings to ensure they would make sense to a translator and have punctuation (if punctuated in English).

Closes #1503

**Notes**
There are probably some unnecessary strings in the .po files that we could get from the node labels, but I didn't want to tackle that refactor now when we have more important follow-ups to get to.
